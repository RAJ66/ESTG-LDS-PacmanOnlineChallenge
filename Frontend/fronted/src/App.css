@import url("https://fonts.googleapis.com/css?family=Roboto:400,700&display=swap");

* {
  margin: 0;
  padding: 0;
  outline: 0;
  box-sizing: border-box;
}

html,
body,
#root {
  min-height: 100%;
}

body,
input,
button {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  font-size: 14px;
}

body,
button {
  background: #0f1d28;
  -webkit-font-smoothing: antialiased !important;
}

.container-inicio {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: #0f1d28;
}

.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: #fff;
  border-radius: 6px;
  padding: 30px;
}

.login-container input {
  margin-top: 20px;
  border: 2px solid #0f1d28;
  border-radius: 4px;
  height: 48px;
  padding: 0 20px;
  font-size: 16px;
  color: #0f1d28;
}

.login-container input::placeholder {
  color: #0f1d28;
}

.login-container button {
  margin-top: 10px;
  border: 0px;
  border-radius: 4px;
  height: 48px;
  width: 100px;
  font-size: 16px;
  font-weight: bold;
  color: #fff;
  cursor: pointer;
}

.glow-on-hover {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: #fff;
  background: #0f1d28;
  cursor: pointer;
  position: relative;
  z-index: 0;
  border-radius: 10px;
}

.glow-on-hover:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: 10px;
}

.glow-on-hover:active {
  color: #0f1d28;
}

.glow-on-hover:active:after {
  background: transparent;
}

.glow-on-hover:hover:before {
  opacity: 1;
}

.glow-on-hover:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: #0f1d28;
  left: 0;
  top: 0;
  border-radius: 10px;
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}

.login-container img {
  height: 100px;
  width: 150px;
}

@media only screen and (max-width: 600px) {
  body,
  button {
    background: #0f1d28;
    -webkit-font-smoothing: antialiased !important;
  }

  .container {
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    background: #0f1d28;
  }

  .login-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 0px;
    background: #fff;
    max-width: 600px;
    width: 100%;
    padding: 30px;
  }

  .login-container input {
    margin-top: 20px;
    border: 2px solid #0f1d28;
    border-radius: 4px;
    height: 48px;
    padding: 0 20px;
    font-size: 16px;
    color: #0f1d28;
  }

  .login-container input::placeholder {
    color: #0f1d28;
  }

  .login-container button {
    margin-top: 10px;
    border: 0px;
    border-radius: 4px;
    height: 48px;
    width: 100px;
    font-size: 16px;
    font-weight: bold;
    color: #fff;
    cursor: pointer;
  }
}
