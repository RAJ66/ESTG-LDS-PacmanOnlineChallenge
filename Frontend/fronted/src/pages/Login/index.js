import React, { useState } from "react";
import api from "../../services/api";
import logo from "../../assets/logo.png";

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [pass, setPass] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(email);
    console.log(pass);

    api.post("/api/users", {
      UserEmail: email,
      Password: pass
    });

    history.push("/dashboard");
  }

  return (
    <form className="login-container" onSubmit={handleSubmit}>
      <img src={logo} alt="It a match" />
      <input
        placeholder="Email"
        value={email}
        onChange={event => setEmail(event.target.value)}
      />
      <input
        placeholder="Password"
        type="password"
        value={pass}
        onChange={event => setPass(event.target.value)}
      />

      <button type="submit" className="glow-on-hover">
        Login
      </button>
      <button type="" className="glow-on-hover">
        Register
      </button>
    </form>
  );
}
