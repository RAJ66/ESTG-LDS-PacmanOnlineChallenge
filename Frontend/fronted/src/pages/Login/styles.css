.container-inicio {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.login-container input {
  margin-top: 20px;
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 20px;
  font-size: 16px;
  color: var(--color-base);
}

.login-container input::placeholder {
  color: var(--color-base);
}

.login-container button {
  margin-top: 10px;
  border: 0px;
  border-radius: 4px;
  height: 46px;
  width: 100px;
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}

.glow-on-hover {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: var(--color-zero);
  background: var(--color-base);
  cursor: pointer;
  position: relative;
  z-index: 0;
  border-radius: 10px;
}

.glow-on-hover:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: 10px;
}

.glow-on-hover:active {
  color: var(--color-base);
}

.glow-on-hover:active:after {
  background: transparent;
}

.glow-on-hover:hover:before {
  opacity: 1;
}

.glow-on-hover:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: var(--color-base);
  left: 0;
  top: 0;
  border-radius: 10px;
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}

.login-container img {
  height: 100px;
  width: 150px;
}

@media only screen and (max-width: 600px) {
  body,
  button {
    background: var(--color-base);
    -webkit-font-smoothing: antialiased !important;
  }

  .container {
    width: 100vw;
    height: 100vh;
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    background: var(--color-base);
  }

  .login-container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 0px;
    background: var(--color-zero);
    max-width: 600px;
    width: 100%;
    padding: 30px;
  }

  .login-container input {
    margin-top: 20px;
    border: 2px solid var(--color-base);
    border-radius: 4px;
    height: 46px;
    padding: 0 20px;
    font-size: 16px;
    color: var(--color-base);
  }

  .login-container input::placeholder {
    color: var(--color-base);
  }

  .login-container button {
    margin-top: 10px;
    border: 0px;
    border-radius: 4px;
    height: 46px;
    width: 100px;
    font-size: 16px;
    font-weight: bold;
    color: var(--color-zero);
    cursor: pointer;
  }
}
