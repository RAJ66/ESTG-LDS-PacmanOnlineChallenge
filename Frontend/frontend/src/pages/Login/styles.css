.container-inicio {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background: var(--color-base);
}

.login-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  max-width: 400px;
  width: 100%;

  background: var(--color-zero);
  border-radius: 6px;
  padding: 30px;
}

.login-container input {
  margin-top: var(--size-small-medium);
  border: 2px solid var(--color-base);
  border-radius: 4px;
  height: 48px;
  padding: 0 var(--size-small-medium);
  font-size: 16px;
  color: var(--color-base);
}

.login-container button {
  margin-top: var(--size-small);
  border: 0px;
  border-radius: 4px;
  height: var(--size-medium);
  width: var(--size-big);
  font-size: 16px;
  font-weight: bold;
  color: var(--color-zero);
  cursor: pointer;
}

.glow-on-hover {
  width: 220px;
  height: 50px;
  border: none;
  outline: none;
  color: var(--color-zero);
  background: var(--color-base);
  cursor: pointer;
  position: relative;
  z-index: 0;
  border-radius: var(--size-small);
}

.glow-on-hover:before {
  content: "";
  background: linear-gradient(
    45deg,
    #ff0000,
    #ff7300,
    #fffb00,
    #48ff00,
    #00ffd5,
    #002bff,
    #7a00ff,
    #ff00c8,
    #ff0000
  );
  position: absolute;
  top: -2px;
  left: -2px;
  background-size: 400%;
  z-index: -1;
  filter: blur(5px);
  width: calc(100% + 4px);
  height: calc(100% + 4px);
  animation: glowing 20s linear infinite;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  border-radius: var(--size-small);
}

.glow-on-hover:active {
  color: var(--color-base);
}

.glow-on-hover:active:after {
  background: transparent;
}

.glow-on-hover:hover:before {
  opacity: 1;
}

.glow-on-hover:after {
  z-index: -1;
  content: "";
  position: absolute;
  width: 100%;
  height: 100%;
  background: var(--color-base);
  left: 0;
  top: 0;
  border-radius: var(--size-small);
}

@keyframes glowing {
  0% {
    background-position: 0 0;
  }
  50% {
    background-position: 400% 0;
  }
  100% {
    background-position: 0 0;
  }
}

.logo {
  height: var(--size-big);
  width: 150px;
}

@media only screen and (max-width: 600px) {
  .login-container {
    border-radius: 0px;
  }
}

@media only screen and (max-height: 550px) {
  .logo {
    height: 75px;
    width: 125px;
  }
  .container-login {
    width: 100vw;
    height: 100vh;
  }

  .login-container {
    width: 100vw;
    height: 100vh;
  }

  .login-container {
    border-radius: 0px;
  }
}
