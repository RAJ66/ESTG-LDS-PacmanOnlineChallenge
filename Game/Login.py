import pygame
import json
import requests
from MenuPrincipal import menuPrincipal

pygame.init()
screen = pygame.display.set_mode((610, 670))
COLOR_INACTIVE = pygame.Color('lightskyblue3')
COLOR_ACTIVE = pygame.Color('dodgerblue2')
FONT = pygame.font.Font(None, 32)


class InputBox:
    def __init__(self, x, y, w, h, n, text=''):
        self.rect = pygame.Rect(x, y, w, h)
        self.color = COLOR_INACTIVE
        self.text = text
        self.txt_surface = FONT.render(text, True, self.color)
        self.active = False
        self.font = pygame.font.Font('PAC-FONT.TTF', 25)
        self.hide = ''
        self.number = n

    def handle_event(self, event):
        if event.type == pygame.MOUSEBUTTONDOWN:
            if self.rect.collidepoint(event.pos):
                self.active = not self.active
            else:
                self.active = False
            self.color = COLOR_ACTIVE if self.active else COLOR_INACTIVE
        if event.type == pygame.KEYDOWN:
            password = ''
            if self.active:
                if event.key == pygame.K_RETURN:
                    self.text = ''
                elif event.key == pygame.K_BACKSPACE:
                    self.text = self.text[:-1]
                    self.hide = self.hide[:-1]
                else:
                    password += '*'
                    self.text += event.unicode
                    self.hide += '*'
                if self.number == 1:
                    self.txt_surface = FONT.render(self.text, True, self.color)
                else:
                    self.txt_surface = FONT.render(self.hide, True, self.color)

    def update(self):
        width = max(200, self.txt_surface.get_width()+10)
        self.rect.w = width

    def draw(self, screen):
        screen.blit(self.txt_surface, (self.rect.x+5, self.rect.y+5))
        pygame.draw.rect(screen, self.color, self.rect, 2)

    def addText(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 25).render(
            'login', True, (255, 0, 0)), (260, 400))

    def addText2(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 25).render(
            'ONLINE CHALLENGE', True, (255, 255, 0)), (150, 250))

    def addText3(self):
        screen.blit(pygame.font.Font('PAC-FONT.TTF', 15).render(
            'CLICK HERE IF YOU FORGOT YOUR ACCOUNT DETAILS', True, (255, 0, 0)), (20, 600))


def main():
    clock = pygame.time.Clock()
    input_box1 = InputBox(220, 300, 140, 32, 1, "Username")
    input_box2 = InputBox(220, 350, 140, 32, 2, "")
    input_boxes = [input_box1, input_box2]
    done = False

    username = None
    password = None

    while not done:

        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                done = True
            if event.type == pygame.KEYDOWN:
                if event.key == pygame.K_RETURN:
                    url = "https://localhost:5001/api/user/auth"
                    data = {'Username': username, 'Password': password}
                    headers = {'Content-type': 'application/json',
                               'Accept': 'text/plain'}
                    r = requests.post(url, data=json.dumps(
                        data), headers=headers, verify=False)
                    if r.status_code == 200:
                        menuPrincipal(username)
                    else:
                        print("USERNAME OU PASSWORD INVALIDA")
                        main()
            for box in input_boxes:
                box.handle_event(event)

        for box in input_boxes:
            box.update()

        screen.fill((0, 0, 0))
        for box in input_boxes:
            box.draw(screen)

        print(username)
        print(password)

        username = input_box1.text.strip()
        password = input_box2.text.strip()

        InputBox.addText3(screen)
        InputBox.addText2(screen)
        button = pygame.Rect(260, 400, 100, 40)
        pygame.draw.rect(screen, [0, 0, 0], button)
        InputBox.addText(screen)
        img = pygame.image.load('img/logo.jpg')
        screen.blit(pygame.transform.scale(img, (610, 250)), (0, 0))
        pygame.display.flip()
        clock.tick(10)


main()
pygame.quit()
