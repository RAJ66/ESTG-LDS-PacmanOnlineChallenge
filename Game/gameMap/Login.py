import pygame as pg

pg.init()
screen = pg.display.set_mode((640, 480))
COLOR_INACTIVE = pg.Color('lightskyblue3')
COLOR_ACTIVE = pg.Color('dodgerblue2')
FONT = pg.font.Font(None, 32)


class InputBox:
    def __init__(self, x, y, w, h, text=''):
        self.rect = pg.Rect(x, y, w, h)
        self.color = COLOR_INACTIVE
        self.text = text
        self.txt_surface = FONT.render(text, True, self.color)
        self.active = False
        self.font = pg.font.SysFont('Arial', 25)

    def handle_event(self, event):
        if event.type == pg.MOUSEBUTTONDOWN:
            if self.rect.collidepoint(event.pos):
                self.active = not self.active
            else:
                self.active = False
            self.color = COLOR_ACTIVE if self.active else COLOR_INACTIVE
        if event.type == pg.KEYDOWN:
            password = ''
            if self.active:
                if event.key == pg.K_RETURN:
                    self.text = ''
                    print("123")
                    print(password)
                elif event.key == pg.K_BACKSPACE:
                    self.text = self.text[:-1]
                else:
                    password += '*'
                    self.text += event.unicode
                self.txt_surface = FONT.render(self.text, True, self.color)

    def update(self):
        width = max(200, self.txt_surface.get_width()+10)
        self.rect.w = width

    def draw(self, screen):
        screen.blit(self.txt_surface, (self.rect.x+5, self.rect.y+5))
        pg.draw.rect(screen, self.color, self.rect, 2)

    def addText(self):
        screen.blit(pg.font.SysFont('PacFont', 25).render(
            'login', True, (255, 0, 0)), (260, 300))


def main():
    clock = pg.time.Clock()
    input_box1 = InputBox(200, 200, 140, 32)
    input_box2 = InputBox(200, 250, 140, 32)
    input_boxes = [input_box1, input_box2]
    done = False

    while not done:
        for event in pg.event.get():
            if event.type == pg.QUIT:
                done = True
            for box in input_boxes:
                box.handle_event(event)

        for box in input_boxes:
            box.update()

        screen.fill((30, 30, 30))
        for box in input_boxes:
            box.draw(screen)

        button = pg.Rect(260, 300, 100, 40)
        pg.draw.rect(screen, [255, 255, 255], button)
        InputBox.addText(screen)
        img = pg.image.load('img/logo.png')
        screen.blit(pg.transform.scale(img, (200, 150)), (200, 10))
        pg.display.flip()
        clock.tick(30)


if __name__ == '__main__':
    main()
    pg.quit()
