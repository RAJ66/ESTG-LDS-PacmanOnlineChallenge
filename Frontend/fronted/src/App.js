import React, { useState } from "react";
import "./App.css";
import api from "./services/api";

import logo from "./assets/logo.png";

function App() {
  const [email, setEmail] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(email);
  }

  return (
    <div className="container">
      <form className="login-container" onSubmit={handleSubmit}>
        <img src={logo} alt="It a match" />
        <input
          placeholder="Email"
          value={email}
          onChange={event => setEmail(event.target.value)}
        />
        <input placeholder="Password" type="password" />

        <button type="submit" className="glow-on-hover">
          Login
        </button>
        <button type="" className="glow-on-hover">
          Register
        </button>
      </form>
    </div>
  );
}

export default App;
