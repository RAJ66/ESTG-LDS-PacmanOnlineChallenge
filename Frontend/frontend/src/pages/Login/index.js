import React, { useState } from "react";
import api from "../../services/api";
import logo from "../../assets/logo.png";
import "./styles.css";

export default function Login({ history }) {
  const [email, setEmail] = useState("");
  const [pass, setPass] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(email);
    console.log(pass);

    api.post("/api/user/auth", {
      username: email,
      password: pass
    });
  }

  function handleClickRegister(event) {
    event.preventDefault();
    history.push("/register");
  }

  function handleClickForgot(event) {
    event.preventDefault();
    alert("Email send");
    history.push("/");
  }

  return (
    <div className="container-inicio">
      <form className="login-container" onSubmit={handleSubmit}>
        <img src={logo} className="logo" alt="It a match" />
        <input
          placeholder="Email"
          value={email}
          onChange={event => setEmail(event.target.value)}
        />
        <input
          placeholder="Password"
          type="password"
          value={pass}
          onChange={event => setPass(event.target.value)}
        />

        <button type="submit" className="glow-on-hover basebutton">
          Login
        </button>
        <button
          type="reset"
          onClick={handleClickRegister}
          className="glow-on-hover basebutton"
        >
          Register
        </button>

        <button onClick={handleClickForgot} className="forgotButton ">
          Forgot account?
        </button>
      </form>
    </div>
  );
}
