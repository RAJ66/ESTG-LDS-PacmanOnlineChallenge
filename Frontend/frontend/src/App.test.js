import React from "react";
import { shallow, mount } from "enzyme";
import App from "./App";
import Login from "./pages/Login/index.js";
import ShallowRenderer from "react-test-renderer/shallow"; // ES6
import toJson from "enzyme-to-json";

describe("Testing Login Component", () => {
  it("should render correctly", () => {
    const wrapper = shallow(
      <div className="my-component">
        <strong>ola World!</strong>
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("knows that 2 and 2 make 4", () => {
    expect(2 + 2).toBe(4);
  });
});
