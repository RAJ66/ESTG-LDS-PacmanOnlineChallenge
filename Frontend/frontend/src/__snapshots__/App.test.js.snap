// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login Component should render correctly 1`] = `
<div
  className="my-component"
>
  <strong>
    ola World!
  </strong>
</div>
`;
